### PR TITLE
Np 1182 create cognito arn parameter

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: src/main/java/no/unit/nva/handlers/GetRoleHandler.java [hunk 1]
```diff
@@ -44,7 +46,9 @@ public GetRoleHandler(Environment environment, DatabaseService databaseService
     }
 
     @Override
-    protected RoleDto processInput(Void input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+    @POST
+    @Path("/roles/{role}")
```
Comment (brinxmat) [inline src/main/java/no/unit/nva/handlers/GetRoleHandler.java]: We should discuss the annotations here

File: src/test/java/features/AddUserFt.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        UserDto updatedRequestBody = currentRequestBody.copy().withRoles(expectedRoles).build();
+
+        HandlerRequestBuilder<Map<String, Object>> requestBuilder = scenarioContext.getRequestBuilder()
+            .withBody(prepareRequestBody(updatedRequestBody));
+        scenarioContext.setRequestBuilder(requestBuilder);
+    }
+
+    @Then("a user description is returned")
+    public void a_user_description_is_returned() throws IOException {
+        resultUser = scenarioContext.getResponseBody(UserDto.class);
```
Comment (brinxmat) [inline src/test/java/features/AddUserFt.java]: Should there be an assertion here?

File: src/test/java/features/GetRoleFt.java [hunk 3]
```diff
@@ -51,19 +59,23 @@ public void that_there_is_a_role_with_rolename(String roleName)
...
+            .withPathParameters(pathParameters));
     }
 
     @Then("a role description is returned")
     public void a_role_description_is_returned() throws IOException {
-        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-        InputStream inputStream = this.requestBuilder.build();
-        GetRoleHandler getRoleHandler = new GetRoleHandler(mockEnvironment(), scenarioContext.getDatabaseService());
-        getRoleHandler.handleRequest(inputStream, outputStream, context);
+        ByteArrayOutputStream outputStream = executeRequest();
```
Comment (brinxmat) [inline src/test/java/features/GetRoleFt.java]: Assertion?

File: src/test/resources/features/Roles.feature [hunk 4]
```diff
@@ -6,22 +6,31 @@ Feature: Role features
     And an authorized client
 
   Scenario: Authorized client creates Role
-    When the authorized client sends a "POST" request
+    Given that the authorized client intends to add a new Role
```
Comment (brinxmat) [inline src/test/resources/features/Roles.feature]: Not sure about the "intends" — is it necessary? "Given that the client is authorized | When the client sends a request with the kv pairs…

File: src/test/resources/features/Roles.feature [hunk 5]
```diff
@@ -6,22 +6,31 @@ Feature: Role features
...
       | rolename | theRole |
+    When the authorized client sends the request
     Then a new role is stored in the database
     And the description of the role is returned to the authorized client
 
-
   Scenario: Authorized client reads Role
-    Given that there is a role with role-name "TheRole"
-    When the authorized client sends a "GET" request with the following path parameters:
+    Given that the authorized client intends to read a Role
```
Comment (brinxmat) [inline src/test/resources/features/Roles.feature]: Ditto

File: src/test/resources/features/Roles.feature [hunk 6]
```diff
@@ -6,22 +6,31 @@ Feature: Role features
...
       | role      | TheRole |
+    When the authorized client sends the request
     Then a role description is returned
-    And the role description contains the following fields and respective values:
-      | field    | fieldValue |
-      | rolename | TheRole    |
-
 
+  Scenario: Authorized client tries to read non-existent Role
+    Given that the authorized client intends to read a Role
```
Comment (brinxmat) [inline src/test/resources/features/Roles.feature:27]: Given given? Same concerns as above

File: src/test/resources/features/Users.feature [hunk 7]
```diff
@@ -1,8 +1,46 @@
...
-#    Given a user with username "someone@institution.com"
-#    When handler receives a request for accessing the user details
-#    Then the handler returns the user details
+  Background:
+    Given a database for users and roles
+    And an authorized client
+
+  Scenario: Authorized client adds a new user
+    Given that the authorized client intends to add a new user
+    And  that a user entry with the username someone@institution does not exist in the database
```
Comment (brinxmat) [inline src/test/resources/features/Users.feature]: Inverted commas for matching purposes

File: src/test/resources/features/Users.feature [hunk 8]
```diff
@@ -1,8 +1,46 @@
...
-#  Scenario: get user details
-#    Given a user with username "someone@institution.com"
-#    When handler receives a request for accessing the user details
-#    Then the handler returns the user details
+  Background:
+    Given a database for users and roles
+    And an authorized client
+
+  Scenario: Authorized client adds a new user
+    Given that the authorized client intends to add a new user
```
Comment (brinxmat) [inline src/test/resources/features/Users.feature]: Intends again

File: build.gradle [hunk 9]
```diff
@@ -58,8 +58,12 @@ dependencies {
     testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: jupiterVersion
     testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-params', version: jupiterVersion
     testImplementation group: 'org.hamcrest', name: 'hamcrest', version: '2.2'
-    runtimeOnly(group: 'io.cucumber', name: 'cucumber-picocontainer', version: '6.1.1') {
-        because("Dependency injection in cucumber glue code")
+    gradleLint.ignore {
+        // gradleLint wants this to be testRuntimeOnly, but cucumber fails to run when this is not
```
Comment (brinxmat) [inline build.gradle:62]: It is a shame there is no "because" for gradleLint.ignore

File: src/test/java/features/GetRoleFt.java [hunk 10]
```diff
@@ -89,10 +87,20 @@ public void the_role_description_contains_the_following_fields_and_respective_va
         compareExpectedAndActualValues(expectedFieldValuePairs, actualFieldValuePairs);
     }
 
+    @When("the authorized client sends the request to read the role")
+    public void theAuthorizedClientSendsTheRequestToReadTheRoleWithRoleName() throws IOException {
```
Comment (brinxmat) [inline src/test/java/features/GetRoleFt.java]: snake and camel naming?

File: src/test/java/features/ScenarioTest.java [hunk 11]
```diff
@@ -20,13 +33,50 @@ protected static DataTable ignoreHeadersRow(DataTable datatable) {
...
+        return scenarioContext.getDatabaseService();
+    }
+
+    private <I, O> ByteArrayOutputStream invokeHandlerWithRequest(
+        ApiGatewayHandler<I, O> handler,
+        InputStream request) throws IOException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        Context context = mock(Context.class);
+        handler.handleRequest(request, outputStream, context);
```
Comment (brinxmat) [inline src/test/java/features/ScenarioTest.java]: Probably just inline the variables

**Review by brinxmat — DISMISSED** [review]

File: src/test/java/features/ScenarioTest.java (hunk 11 above)
Comment (axthosarouris) [reply src/test/java/features/ScenarioTest.java]: only context can be inlined. outputStream in returned

File: src/test/java/features/GetRoleFt.java (hunk 10 above)
Comment (axthosarouris) [reply src/test/java/features/GetRoleFt.java]: different tools. fixed.

**Review by brinxmat — APPROVED** [review]

**Pull request merged**